src/lib/addons/slack-app.ts

```diff
@@ -17,64 +21,103 @@ interface ISlackAppAddonParameters {
+                );
 
-        const slackChannels = await this.slackClient.conversations.list({
-            types: 'public_channel,private_channel',
-        });
-        const taggedChannels = this.findTaggedChannels(event);
-
-        if (slackChannels.channels?.length && taggedChannels.length) {
-            const slackChannelsToPostTo = slackChannels.channels.filter(
-                ({ id, name }) => id && name && taggedChannels.includes(name),
-            );
-
-            const text = this.msgFormatter.format(event);
-            const featureLink = this.msgFormatter.featureLink(event);
-
-            const requests = slackChannelsToPostTo.map(({ id }) =>
-                this.slackClient!.chat.postMessage({
-                    channel: id!,
-                    text,
-                    attachments: [
-                        {
-                            actions: [
-                                {
-                                    name: 'featureToggle',
-                                    text: 'Open in Unleash',
-                                    type: 'button',
-                                    value: 'featureToggle',
-                                    style: 'primary',
-                                    url: featureLink,
-                                },
-                            ],
-                        },
-                    ],
-                }),
-            );
-
-            await Promise.all(requests);
-            this.logger.info(`Handled event ${event.type}.`);
+                const text = this.msgFormatter.format(event);
+                const featureLink = this.msgFormatter.featureLink(event);
+
+                const requests = slackChannelsToPostTo.map(({ id }) =>
+                    this.slackClient!.chat.postMessage({
+                        channel: id!,
+                        text,
+                        attachments: [
+                            {
+                                actions: [
+                                    {
+                                        name: 'featureToggle',
+                                        text: 'Open in Unleash',
+                                        type: 'button',
+                                        value: 'featureToggle',
+                                        style: 'primary',
+                                        url: featureLink,
+                                    },
+                                ],
+                            },
+                        ],
+                    }),
+                );
+
+                await Promise.all(requests);
+                this.logger.info(`Handled event ${event.type}.`);
+            }
+        } catch (error) {
+            if (error.code === ErrorCode.PlatformError) {
+                this.logger.error(
+                    `Error handling event ${event.type}. A platform error occurred: ${error.data}`,
+                    error,
+                );
```

Comment:
Should these errors be "errors"? I don't think we can do much (as SRE oncalls) if there's a surge of errors because SlackAPI is down... I think a warning would be better here

Reply:
Yep, addressed in https://github.com/Unleash/unleash/pull/4284/commits/1d2db10de70b3c2b1cdb3bda3df57cb9b6683131